### ISSUE TITLE
Search by vol/page reference

### ISSUE THREAD

**Issue by sujato** [suttacentral/suttacentral#1367]
## User story

Janice wants to look up a reference from an academic article. The footnote says "M ii 267". She writes that in the SC search field, but reaps only bitterness and disappointment. Frustrated, she turns to drink, looking for answers in a bottle of tequila when she should be looking for them on SuttaCentral.

## Feature description

One should be able to input a vol/page reference in the search field and return the sutta where that page occurs. 

It's particularly important to return results for the PTS and Taisho vol/page references.

Data for references is sourced from bilara-data/reference. (This applies to all Pali and some Chinese texts. Build from this and let the number of texts grow over time.)

This feature is provided by an external service: http://pts.ticao.de/

One problem to be dealt with is the aliasses or simply different formatting of such references. In the example above I use M ii 267, but on SC that is MN ii 267. Do we attempt to programmatically fix malformed entries (hard, error-prone), or do we try to get our users to use the correct form? This is a bit more work up front, but I'm thinking, some one who is using this feature is a serious researcher, so they should be able to stand a bit of a learning curve.

Perhaps a prompt or explainer for the correct form?

## Acceptance criteria

- [x] Search recognizes vol/page references
- [x] References within suttas are correlated with the appropriate segment number.
- [x] Search results return the text, with the relevant page highlighted.

**Comment by Manfredsuttacentral** [comment]
Is there a way to query SC for links corresponding to pages of the PTS-Pali editions, for example something like  query: "A. i. 123" ->  response: "https://suttacentral.net/an1.21-30"? Or maybe a list could be compiled? this feature would come in handy for batch-linking references in footnotes of PTS-digital editions.

**Comment by sujato** [comment]
The correlation between PTS and other edition reference numbers and SC references is now systematically mapped in our bilara-data/reference files. These could possibly be downloaded into memory for instant search, or at least a portion of them.

If you want to batch-link references, use these files. In fact that's what we're planning for the Dictionaries:

https://github.com/suttacentral/suttacentral/issues/2358

**Comment by thesunshade** [comment]
@ihongda This has become very buggy/broken.

The first three examples from the filters page are completely broken (as I understand the feature)

* `volpage:SN ii 4`
* `volpage:M II 246`
* `volpage:S.II,236`

They are showing suttas from throughout the canon.

The fourth example, `volpage:SN ii` has a first result of this:

![Image](https://github.com/suttacentral/suttacentral/assets/82448383/fa9d2f81-a6db-4dd0-8d76-c9259ec60521)


It takes you to the link https://suttacentral.net/an3.29_sn22.57 which isn't a real thing as far as I know.

Also, I notice that (at least for the fifth example https://suttacentral.net/search?query=volpage:M%20II) The results are sorted "alphabetically" with the MN 100's coming before MN 77+

**Comment by thesunshade** [comment]
I'm noticing another bug, although I can't figure out what is going on.

Searching for DN ii 132 gives a blank page: https://staging.suttacentral.net/search?query=volpage:DN%20ii%20132

![image](https://github.com/suttacentral/suttacentral/assets/82448383/b73ce53f-023f-4b5a-a559-e95d24bf1416)

That citation exists, though: https://staging.suttacentral.net/dn16/en/sujato?lang=en&layout=plain&reference=pts&notes=asterisk&highlight=false&script=latin#pts-vp-pli2.132

**Comment by thesunshade** [comment]
@ihongda just wanted to make sure you knew this is completely broken now. 

I just needed to look up `DN iii 66`. I tried `D iii 66` and in both case now I get a no results returned page on staging:

![Image](https://github.com/suttacentral/suttacentral/assets/82448383/e6a91f4f-767e-4fd2-8507-fee547c5500e)

Same on production.

This is rather important since the current Pali class was using a lot of DN texts.
